### PR TITLE
New Hampton Bay fan controller #1357

### PR DESCRIPTION
**Pull request by dwest576**

### CODE REVIEW

File: devices.js [hunk 1]
```diff
@@ -138,6 +138,42 @@ const execute = (device, actions, callback, delay) => {
...
+        zigbeeModel: ['HDC52EastwindFan'],
+        model: '99432',
+        vendor: 'Hampton Bay',
+        description: 'Universal wink enabled white ceiling fan premier remote control',
+        supports: 'on/off',
+        fromZigbee: [
+            fz.generic_state, fz.light_brightness, fz.ignore_fan_change, fz.generic_fan_mode, fz.ignore_onoff_change,
+            fz.ignore_level_report,
+        ],
+        toZigbee: [tz.on_off, tz.light_onoff_brightness, tz.light_brightness, tz.fan_mode],
```
Comment (Koenkk) [inline devices.js]: I think these should be removed: `tz.light_onoff_brightness, tz.light_brightness`.

**Comment by Koenkk** [comment]
Are you using Home Assistant?

File: devices.js (hunk 1 above)
Comment (dwest576) [reply devices.js]: I am not sure if tz.light_onoff_brightness is needed, but tz.light_brightness should remain as the fan does have a light attached and the brightness command does function.

**Comment by dwest576** [comment]
> Are you using Home Assistant?

Yes, just haven't gotten to that part of the integration yet.   Also, not running Z2M on the hass devices.

File: devices.js (hunk 1 above)
Comment (Koenkk) [reply devices.js]: Ok, I wasn't aware of the light, about the on/off converter, is it used to turn on/off the fan or the light? (meaning that the `fan_mode` converter is only used to control the fan).

Comment (dwest576) [reply devices.js]: I believe that tz.on_off controls just the light.  Assuming when I pass "state: off" to qitt that is sends a tz.on_off command.  When I have controlled the fan it has always been through fan_mode.  There is a button on the remote that comes with the controller itself, that is a "all" power button, which turns off the light and the fan all at once.  I have never used it, but it is there.

File: devices.js [hunk 2]
```diff
@@ -138,6 +138,42 @@ const execute = (device, actions, callback, delay) => {
...
 const devices = [
+    // Hampton Bay Fan Controller
+    {
+        zigbeeModel: ['HDC52EastwindFan'],
+        model: '99432',
+        vendor: 'Hampton Bay',
+        description: 'Universal wink enabled white ceiling fan premier remote control',
+        supports: 'on/off',
+        fromZigbee: [
+            fz.generic_state, fz.light_brightness, fz.ignore_fan_change, fz.generic_fan_mode, fz.ignore_onoff_change,
```
Comment (Koenkk) [inline devices.js]: The `generic_fan_mode` converter is missing, can you also add it?

**Comment by dwest576** [comment]
This looks awesome, anything else you need from me.  I am happy to test everything and report back.

**Comment by Koenkk** [comment]
@dwest576 could you provide me the `fz.generic_fan_mode`? It's missing from the changed files.

File: devices.js (hunk 2 above)
Comment (dwest576) [reply devices.js]: the device insert was from https://github.com/Koenkk/zigbee2mqtt/issues/1357

I'll reach out to the author for that code.

**Comment by ccalica** [comment]
Thanks for doing this.  I've been eyeing this controller for awhile.  Just ordered, will test/help when it arrives.

**Comment by Koenkk** [comment]
This should be ready to merge now, @dwest576 can you do some final tests to ensure everything still works?

**Comment by dwest576** [comment]
> This should be ready to merge now, @dwest576 can you do some final tests to ensure everything still works?

I'll test it when I get home today, do I need to rebuild or simply place the changed files in the proper folder?

**Comment by Koenkk** [comment]
just replace the changed files.

**Comment by dwest576** [comment]
> just replace the changed files.

I tested this evening.   Everything works great.   I paired the device, it pulls everything in, no errors.  I press all the buttons on the remote and I see all the changes in the z2m log.   I sent many commands from mqtt to z2m and it passes them all via zigbee to the fan it all works.

The only thing that doesn't work yet is the auto config into homeasisstant.   I replaced homeassistant.js with the new code and when I pari the fan it doesn't do anything to ha.

**Comment by Koenkk** [comment]
I've integrated this in the zigbee2mqtt dev branch, meaning that home assistant discovery should also work there. Please let me know :smile:

**Comment by dwest576** [comment]
At this point, can I just do a new install and pull all this in and test?

**Comment by Koenkk** [comment]
See http://www.zigbee2mqtt.io/how_tos/how-to-switch-to-dev-branch.html

**Comment by dwest576** [comment]
I was able to switch to the dev branch and the fan now gets pushed to homeassistant.   Would it be possible to get the light pushed to homeassistant too?


I did test all the options from homeassistant to the fan and they all work great!

On April 26, 2019 at 11:12 PM, Koen Kanters <notifications@github.com> wrote:



See http://www.zigbee2mqtt.io/how_tos/how-to-switch-to-dev-branch.html

**Comment by Koenkk** [comment]
Forgot to add that, added it know :smile:

**Comment by dwest576** [comment]
Just updated Dev, and it is now pushing light to Home Assistant.   I’ll test this evening and make sure it works as intended.   Thanks for all your help!

**Pull request merged**